gap/main/setup.gi

```diff
@@ -12,8 +12,11 @@ SEMIGROUPS.HashFunctionRZMSE := function(x, data, func)
   if x![1] = 0 then
     return 1;
   fi;
-
-  return (x![1] + x![3] + func(x![2], data)) mod data + 1;
+  if IsNBitsPcWordRep(x![2]) then
+    return (x![1] + x![3] + func(x![2], data)) mod data[2] + 1;
+  else
```

Comment:
It would probably be better to do 

    elif IsPerm(x![2]) then 
       return ...
    else 
       ErrorNoReturn("Semigroups: SEMIGROUPS.HashFunctionRZMSE: error,\n",
                                 " this shouldn't happen");
    fi;

This way it will give a proper error message if this crops up again in the future.